source/slang-glslang/slang-glslang.cpp

```diff
@@ -145,6 +145,28 @@ struct SPIRVOptimizationDiagnostic
     std::string message;
 };
 
+// TODO: the actual printing should happen on the application side.
```

Comment:
We should extend the `DownstreamCompiler::validate` interface to return a IBlob for the diagnostics.

Reply:
Right. I was little bit reluctant to implement it with IBlob due to the complexity it may come with.
Alternatively, we can take a call-back function, which can avoid the memory allocation and IBlob.

This printing is only for slang developers and printing the message directly from slang-glslang will be fine as a quick fix.
I will merge the change as it for now.